### DISCUSSION TITLE
Format Webhook Message Response Tidak sesuai dokumentasi

### DISCUSSION THREAD

**Discussion by KAnggara75** [kirimwa-id/forum#149] [Tanya Jawab] [answered]
Terdapat perbedaan format pengiriman data pada Webhook Message Response di dokumntasi dan yg saya terima, mohon untuk di cek

format data dari dokumentasi sebagai berikut:

```json
{
	"id": "kwid-426564a5db7940288dc9fddb812",
	"webhook_type": "send_message_response",
	"status": "success",
	"message": "Message has been sent.",
	"payload": {
		"message": "Halo ini adalah pesan dari api.kirimwa.id",
		"phone_number": "6281234567890",
		"device_id": "asus-zenfone-1",
		"message_type": "text",
		"caption": null
	},
	"created_at": "2021-07-10T00:16:10.373Z",
	"server_time": "2021-07-19T13:35:00.078Z"
}
```

tapi data yg saya dapatkan adalah sebagai berikut:
```json
{
	"id": "kwid-f6cca5f567ad407a9d2e1f7c942",
	"status": "success",
	"webhook_type": "send_message_response",
	"payload": {
		"message": "*Daya melebihi batas maksimal* \n\n1818 VA",
		"phone_number": "62822847*****",
		"device_id": "admina",
		"message_type": "text",
		"caption": null,
		"is_group_message": false,
		"send_at": "now"
	},
	"created_at": "2022-11-04T16:28:54.014Z",
	"server_time": "2022-11-04T16:28:54.371Z"
}
```

tidak ada "message": "Message has been sent.",

**Comment by rioastamal** [comment]
Terima kasih laporannya. Saya telah menghapus nomor HP dari posting karena itu informasi sensitif. Lain kali mohon diperhatikan.

Saya tidak menemukan masalah serupa ketika melakukan pengetesan.

```json
{
  "id": "kwid-6d94ff76cab24b02932076d84ee",
  "message": "Message has been sent.",
  "status": "success",
  "created_at": "2022-11-06T05:20:00.671Z",
  "payload": {
    "message": "https://upload.wikimedia.org/wikipedia/commons/thumb/0/04/Terraform_Logo.svg/640px-Terraform_Logo.svg.png",
    "phone_number": "62853355*****-1628639662",
    "device_id": "asus-zenfone-1",
    "message_type": "image",
    "caption": "Terraform Sun Nov  6 12:20:00 WIB 2022",
    "is_group_message": true,
    "send_at": "now"
  }
}
```

Apakah semuanya demikian atau hanya pesan ini saja? Silahkan dicoba lagi.

**Comment by KAnggara75** [comment, marked as answer]
webhook "send_message_response", yang saya terima masih sama seperti sebelumnya, 

```json
{
	"id": "kwid-cd4eeff9f59d42f0b13e2ee73e2",
	"status": "success",
	"webhook_type": "send_message_response",
	"payload": {
		"message": "Halo ini adalah pesan dari api.kirimwa.id 3",
		"phone_number": "62822********",
		"device_id": "admina",
		"message_type": "text",
		"caption": null,
		"is_group_message": false,
		"send_at": "now"
	},
	"created_at": "2022-11-06T05:58:38.577Z",
	"server_time": "2022-11-06T05:58:40.173Z"
}

```
saya menyimpan data webhook ke db, dengan cara seperti berikut

```php
public function debug()
{
	$req = request()->all();
	$data = json_encode($req);
	$webhook = new Webhook();
	$webhook->data = $data;
	$webhook->save();

	return response()->json($webhook,201,[],JSON_NUMERIC_CHECK);
}
```

Terimakasih sebelumnya

**Reply by rioastamal** [reply]
Coba gunakan cURL atau HTTP client lain secara langsung.

```sh
curl 'https://api.kirimwa.id/v1/messages/kwid-cd4eeff9f59d42f0b13e2ee73e2' \
  -H 'Authorization: Bearer API_TOKEN'
```

**Comment by dp20bp** [comment]
mas @rioastamal, response webhook "incoming_message" yang bisa tampil payload_message nya hanya  yang dari hasil KirimWA.id saja kah? Jika bukan dari id kwid-*, payload_message nya tidak ada.

info: saya menampilkan pakai Add-ons webhooks di Google Sheets.

**Reply by rioastamal** [reply]
1. Silahkan update webhook URL sementara ke situs yang bisa menampilkan webhook secara realtime untuk debugging misal https://webhook.site/. 
2. Pastikan properti `allowed_senders` tidak memfilter nomor tertentu

**Reply by dp20bp** [reply]
Berikut POST /webhooks yang saya gunakan untuk mendaftarkan webhook 

curl --location --request POST 'https://api.kirimwa.id/v1/webhooks' \\
--header 'Authorization: Bearer [TOKEN-KIRIMWA.ID]’ \\
--header 'Content-Type: application/json' \\
--data-raw '
{
  "webhook_url": "https://script.google.com/macros/s/[TOKEN-WEBHOOK.SITE]/exec?gid=[ID-WORKSHEET]”
}'




<a href="https://ibb.co/gT776v4"><img src="https://i.ibb.co/3NsskfR/Screenshot-2023-02-13-at-10-27-56-PM.png" alt="Screenshot-2023-02-13-at-10-27-56-PM" border="0" /></a>
Screenshot penerapan pada Google spreadsheet yang saya gunakan ( image akan dihapus sendirinya setelah 6 hari dari sekarang)


payload “allowed_senders” tidak dimasukkan apakah default sudah sebagai wild-card ya mas Rio?

**Reply by rioastamal** [reply]
Benar, jika `allowed_sender` tidak dimasukkan maka default adalah wild card (semua sender akan diproses)

**Reply by dp20bp** [reply]
Baik, terima kasih.
Saya ternyata kurangnya di kolom pada GoogleSheet. payload dari KirimWA.id dan External WA beda payload.
terima kasih mas Rio, sudah solve
